alvr/filesystem/src/lib.rs

```diff
@@ -0,0 +1,200 @@
+use std::{
+    env,
+    path::{Path, PathBuf},
+};
+
+#[cfg(target_os = "linux")]
+const SERVER_BUILD_DIR_NAME: &str = "alvr_server_linux";
+#[cfg(windows)]
+const SERVER_BUILD_DIR_NAME: &str = "alvr_server_windows";
+#[cfg(target_os = "macos")]
+const SERVER_BUILD_DIR_NAME: &str = "alvr_server_macos";
+
+#[cfg(not(windows))]
+pub fn exec_fname(name: &str) -> String {
+    name.to_owned()
+}
+#[cfg(windows)]
+pub fn exec_fname(name: &str) -> String {
+    format!("{}.exe", name)
+}
+
+#[cfg(target_os = "linux")]
+pub fn dynlib_fname(name: &str) -> String {
+    format!("lib{}.so", name)
+}
+#[cfg(windows)]
+pub fn dynlib_fname(name: &str) -> String {
+    format!("{}.dll", name)
+}
+#[cfg(target_os = "macos")]
+pub fn dynlib_fname(name: &str) -> String {
+    format!("lib{}.dylib", name)
+}
+
+pub fn target_dir() -> PathBuf {
+    Path::new(env!("OUT_DIR")).join("../../../..")
+}
+
+pub fn workspace_dir() -> PathBuf {
+    Path::new(env!("CARGO_MANIFEST_DIR"))
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .into()
+}
```

Comment:
Return a `Result` so the unwraps can turn into `?`s?

Reply:
I changed my mind about using  `trace_err!()` everywhere. If something is supposed to never fail then it makes sense to call unwrap, usually when there is no interaction with sockets or the filesystem. In this case we are working with a filesystem path, but this is a function for xtask, where if it crashes the user can refer to the error on the terminal to diagnose the problem.

Reply:
That's okay, but It makes the code ugly like [callback hell](http://callbackhell.com/).